zetaclient/chains/bitcoin/bitcoin.go

```diff
@@ -224,3 +216,55 @@ func (b *Bitcoin) outboundLogger(id string) *zerolog.Logger {
 
 	return &l
 }
+
+// inboundSkipper returns a skipper function for inbound observation.
+func inboundSkipper(b *Bitcoin, app *zctx.AppContext) bool {
+	isSupported := b.observer.ChainParams().IsSupported
+	isInboundEnabled := app.IsInboundObservationEnabled()
+	isMempoolCongested := app.IsMempoolCongested()
+
+	if !isSupported || !isInboundEnabled || isMempoolCongested {
+		b.observer.Logger().
+			Chain.Debug().
```

Comment:
Do you think we should use info? I don't know if debug is currently used for our clients. 
Atleast intitially when we want to see how this new logic is behaving 

it might cause log spam though so we might need to add some condition as to how often this log is printed

Reply:
This commit switched to sampled log https://github.com/zeta-chain/node/pull/4291/commits/28d6fe8611e5023babc878576f423e67b9420699